sympy/integrals/meijerint.py

```diff
@@ -293,7 +293,6 @@ def _mytype(f, x):
         res = []
         for t in types:
             res += list(t)
-        res.sort()
```

Comment:
If anyone understands this function better and has a better suggestion, let me know (see the commit message). Basically, we can't sort `res` because `res` is a list of classes, and ordering of classes is what is being removed here. 

The whole function itself seems a bit shaky, and indeed, all the various ways I've tried to change it result in completely different (but I think mathematically equivalent) results from meijerint.